Source/Classes/PINRemoteImageManager.m

```diff
@@ -881,15 +889,26 @@ + (BOOL)retriableError:(NSError *)remoteImageError
 
 - (NSURLSessionDataTask *)downloadDataWithURL:(NSURL *)url
                                           key:(NSString *)key
+                                   resumeData:(PINResume *)resume
                                      priority:(PINRemoteImageManagerPriority)priority
                                    completion:(PINRemoteImageManagerDataCompletion)completion
 {
     NSMutableURLRequest *request = [NSMutableURLRequest requestWithURL:url
-                                             cachePolicy:NSURLRequestReloadIgnoringLocalCacheData
-                                         timeoutInterval:self.timeout];
-    if (self.httpHeaderFields.count > 0) {
-        request.allHTTPHeaderFields = [self.httpHeaderFields copy];
+                                                           cachePolicy:NSURLRequestReloadIgnoringLocalCacheData
+                                                       timeoutInterval:self.timeout];
+    
+    NSMutableDictionary *headers = [self.httpHeaderFields mutableCopy];
+    
+    if (resume) {
+        headers[@"If-Range"] = resume.ifRange;
```

Comment:
Should we check here if the `ifRange` is null? I think it's declared as `nullable`.